src/services/expenses.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+let expenses = [];
+
+function initiate(initiateValue) {
+  expenses = initiateValue;
+}
+
+function getAll(query) {
+  let filteredExpenses = expenses;
+
+  const { userId, categories, from, to } = query;
+
+  const categoriesArr = Array.isArray(categories)
+    ? categories
+    : [categories];
+
+  if (userId) {
+    filteredExpenses = filteredExpenses.filter(expense => (
+      expense.userId === +userId
+    ));
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter(expense => (
+      categoriesArr.includes(expense.category)
+    ));
+  }
+
+  if (from) {
+    filteredExpenses = filteredExpenses.filter(expense => (
+      Date.parse(expense.spentAt) > Date.parse(from)
+    ));
+  }
+
+  if (to) {
+    filteredExpenses = filteredExpenses.filter(expense => (
+      Date.parse(expense.spentAt) < Date.parse(to)
+    ));
+  }
+
+  return filteredExpenses;
+}
+
+function getById(expenseId) {
+  const foundExpense = expenses.find(expense => expense.id === +expenseId);
+
+  return foundExpense || null;
+}
+
+function create(userId,
+  spentAt,
+  title,
+  amount,
+  category,
+  note,
+) {
+  const id = !expenses.length
+    ? 0
+    : Math.max(...expenses.map(expense => expense.id)) + 1;
```

Comment:
```suggestion
  const id = !expenses.length
    ? 1
    : Math.max(...expenses.map(expense => expense.id)) + 1;
```
I think that `id` better start from `1`